src/librustc_privacy/lib.rs

```diff
@@ -239,26 +482,26 @@ impl<'a, 'tcx> Visitor<'tcx> for EmbargoVisitor<'a, 'tcx> {
             hir::ItemKind::Use(..) => {}
             // The interface is empty.
             hir::ItemKind::GlobalAsm(..) => {}
-            hir::ItemKind::Existential(hir::ExistTy { impl_trait_fn: Some(_), .. }) => {
-                if item_level.is_some() {
-                    // Reach the (potentially private) type and the API being exposed.
-                    self.reach(item.id).ty().predicates();
-                }
+            hir::ItemKind::Existential(..) => {
+                // FIXME: This is some serious pessimization intended to workaround deficiencies
```

Comment:
does this fix a bug?

Reply:
No, pre-existing (introduced in https://github.com/rust-lang/rust/pull/53545).